### PR TITLE
docs: Improved descriptions and schema adjustments for API references

### PR DESCRIPTION
**Pull request by shinjigi**
### What

This pull request updates the API reference documentation files (docs/api/ref/schemas/**/*.yaml) to:

- Improve clarity and detail by adding descriptions and comments to properties and references.
- Address potential issues related to rapidoc and generator errors.
- Make adjustments to schema definitions for better accuracy.
- Added recursive references for Ingredient and Nutrient schemas in api.yml to fix generator errors.
- Reorganized the structure of the image.yaml, image_role.yaml, image_urls.yaml, nutrients.yaml, product_base.yaml, product_ecoscore.yaml, product_extended.yaml, and product_tags.yaml files for better clarity and maintainability.
- Updated pattern properties and refactored the structure of these YAML files to ensure consistency and correctness.

We believe these changes will enhance the usability and understanding of the API reference for developers.

Please review the changes and let us know if you have any feedback.

### SCREENSHOT

image

### CODE REVIEW

File: docs/api/ref/schemas/image.yaml [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
+      type: object
+      title: image
+      description: |
+        This object represent an image that was uploaded to a product.
+        "imgid" is an integer which is a sequential number unique to each picture.
+      properties:
+        sizes:
+          $ref: "#/components/schemas/Sizes"
+        uploaded_t:
+          type: string
```
Comment (stephanegigandet) [inline docs/api/ref/schemas/image.yaml:32]: all *_t timestamps should be integers in fact

File: docs/api/ref/schemas/product_extended.yaml [hunk 2]
```diff
@@ -101,25 +101,19 @@ properties:
...
+    additionalProperties:
+      type: array
+      items:
         type: array
+        # minItems: 2
+        # maxItems: 2
         items:
-          # same as above
-          $ref: "#/properties/nova_groups_markers/properties/3/items"
+          description: This array has two element for each marker. One
```
Comment (stephanegigandet) [inline docs/api/ref/schemas/product_extended.yaml:115]:
```suggestion
          description: This array has two elements for each marker. The first element is the tag type of the marker (e.g. "ingredients", "additives", "categories"), and the second is the tag id in the taxonomy (e.g. "en:e471")
```

**Comment by stephanegigandet** [comment]
Thank you @shinjigi ! I added a couple of suggestions.

File: docs/api/ref/schemas/image.yaml (hunk 1 above)
Comment (shinjigi) [reply docs/api/ref/schemas/image.yaml:32]: image

You're right! But they are currently returned as "false" strings... Is there a change planned in the immediate future? If we put it in multi-type the generator currently gets stuck.

**Review by void0x23 — APPROVED** [review]

**Review by void0x23 — APPROVED** [review]

File: docs/api/ref/schemas/image.yaml (hunk 1 above)
Comment (stephanegigandet) [reply docs/api/ref/schemas/image.yaml:32]: they were integer in the products I tested, but it's something we have a lot of issues with, there's no type in Perl, so the JSON output can have integers converted to strings depending on how they were used last

**Review by stephanegigandet — APPROVED** [review]

**Pull request merged**